datafusion/optimizer/src/unwrap_cast_in_comparison.rs

```diff
@@ -15,8 +15,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
-//! Pre-cast literal binary comparison rule can be only used to the binary comparison expr.
-//! It can reduce adding the `Expr::Cast` to the expr instead of adding the `Expr::Cast` to literal expr.
+//! Unwrap-cast binary comparison rule can be used to the binary/inlist comparison expr now, and other type
+//! of expr can be added if needed.
+//! This rule can reduce adding the `Expr::Cast` the expr instead of adding the `Expr::Cast` to literal expr.
```

Comment:
```suggestion
//! Unwrap-cast binary comparison rule can be used to  remove `cast` functions
//! from column references.
//!
//! The reason for doing so is that it is often much more efficient to 
//! evaluate a predicate like `c1 < literal` via specialized kernels or pruning 
//! rather than `cast(c1 as TYPE) < literal` 
```